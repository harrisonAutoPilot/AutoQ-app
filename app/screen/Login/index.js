import React, { useState, useEffect, useCallback } from "react";
import LinearGradient from 'react-native-linear-gradient';
import {
    View, Text, Image, TouchableOpacity, Keyboard,
    TouchableWithoutFeedback, ImageBackground,
    StatusBar, ScrollView
} from "react-native";
import Icon from 'react-native-vector-icons/Feather';
import { useSelector, useDispatch } from "react-redux";
import Toast from 'react-native-toast-message';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'

import styles from "./style";
import { AuthBtn as Btn, FormikValidator, InputField } from "@Component/index";
import { login } from "@Request/Auth";
import { cleanup } from "@Store/Auth";
import { loginSchema } from "@Helper/Schema";
import Loader from "@Screen/Loader";
import RetrievePin from "@Screen/RetrievePin"

const Login = (props) => {
    const dispatch = useDispatch();
    const [errMsg, setErrMsg] = useState("");
    const [pinVisibility, setPinVisibility] = useState(true);
    const [loader, setLoader] = useState(false);
    const [isKeyboardVisible, setKeyboardVisible] = useState(false);
    const [showRetrieve, setShowRetrieve] = useState(false);

    const { status, errors } = useSelector((state) => state.auth);

    const loginState = {
        phone: "",
        password: ""
    };

    useEffect(() => {
      
        const keyboardDidShowListener = Keyboard.addListener(
            'keyboardDidShow',
            () => {
                setKeyboardVisible(true); // or some other action
            }
        );
        
        const keyboardDidHideListener = Keyboard.addListener(
            'keyboardDidHide',
            () => {
                setKeyboardVisible(false); // or some other action
            }
        );

        return () => {
            dispatch(cleanup())
            keyboardDidHideListener.remove();
            keyboardDidShowListener.remove();
        }
    }, []);

    const toastConfig = {

        error: () =>
        (
            <View style={[globalStyles.errMainView3]}>
                <Text style={globalStyles.failedResponseText}>{errMsg}</Text>
            </View>
        )
    };

    const wait = (timeout) => {
        return new Promise(resolve => setTimeout(resolve, timeout));
    };

    const waitTime = useCallback((msg) => {
        wait(1000).then(() => {
            setLoader(false);
            setErrMsg(msg);
            Toast.show({
                type: 'error',
                visibilityTime: 5000,
                autoHide: true,
                position: 'top',
                topOffset: 0
            })
        });

    }, []);

    useEffect(() => {
        if (status === "failed" && props.navigation.isFocused()) {
            waitTime(errors?.msg);
        }
    }, [status]);

    const dismissKeyboard = () => Keyboard.dismiss();

    const showPin = () => setPinVisibility(!pinVisibility)

    const submit = async (data) => {
        setErrMsg("");
        setLoader(true)
        await dispatch(login(data));

    };

    const redirectToForgotPin = () => {
        props.navigation.navigate('ForgotPin')
    };

    return (
        <View style={styles.mainBody}>
            <StatusBar hidden />
            <View style={styles.agentCover}>
                <ImageBackground style={styles.agentImg} source={require("@Assets/image/AgentElipse.png")} >
                    <LinearGradient colors={['#B1FF5C', 'transparent']} style={{ backgroundColor: 'transparent', position: 'absolute', top: 0, bottom: 0, left: 0, right: 0, }} />
                </ImageBackground>
                <View style={styles.blueCover}>
                    <Image style={styles.blueImg} source={require("@Assets/image/blueImg.png")} />
                    <View style={styles.innerCover}>
                        <Image source={require("@Assets/image/rh_logo_splashscreen.png")} style={styles.logoImg} />

                        <View style={styles.textCover}>
                            <Text style={styles.smText}>Remedial Health</Text>
                            <Text style={styles.bgText}>AGENT</Text>
                        </View>
                    </View>
                </View>

            </View>

            <View style={[styles.bottomCover, isKeyboardVisible ? styles.keyboardUp : null]}>

                <View style={styles.loginTextCover}>
                    <Text style={styles.loginText}>Account Log In</Text>
                </View>
                <View style={styles.errCover}>
                    {errMsg ? <Toast config={toastConfig} /> : null}
                </View>
                <ScrollView showsVerticalScrollIndicator={false} contentContainerStyle={styles.scrollStyle}>
                    <KeyboardAwareScrollView>
                        <View style={styles.inputCover}>

                            <TouchableWithoutFeedback onPress={dismissKeyboard}>
                                <View>

                                    <FormikValidator
                                        initialValues={loginState}
                                        validationSchema={loginSchema}
                                        onSubmit={(values) => {
                                            submit(values)
                                        }}>
                                        {props => (
                                            <View>
                                                <View>
                                                    <View style={[styles.inputHolder, props.touched.phone && props.errors.phone ? styles.inputErrHolder : null]}>
                                                        <View style={styles.labelView}>
                                                            <Text style={styles.label}>PHONE NUMBER</Text>
                                                        </View>

                                                        <InputField
                                                            value={props.values.phone}
                                                            onBlur={props.handleBlur('phone')}
                                                            placeholder="234809XXXXXXX"
                                                            placeholderTextColor="#757575"
                                                            keyboardType="number-pad"
                                                            onChangeText={(val) => {
                                                                props.setFieldValue('phone', val)
                                                                props.setFieldTouched('phone', true, false);
                                                                setErrMsg("")
                                                            }}
                                                            style={styles.label2}
                                                            autoFocus={true}
                                                            maxLength={13}
                                                        />
                                                    </View>
                                                    {props.touched.phone && props.errors.phone ? (
                                                        <View style={styles.errView}>
                                                            <Text style={styles.errText}>{props.errors.phone}</Text>
                                                        </View>) : null}
                                                </View>

                                                <View>
                                                    <View style={[styles.inputHolder, styles.inputPinHolder, props.touched.password && props.errors.password ? styles.inputErrHolder : null]}>
                                                        <View style={styles.labelView}>
                                                            <Text style={styles.label}>4 DIGIT PIN</Text>
                                                        </View>
                                                        <View style={styles.pinInputView}>
                                                            <InputField
                                                                value={props.values.password}
                                                                onBlur={props.handleBlur('password')}
                                                                placeholder="****"
                                                                placeholderTextColor="#757575"
                                                                keyboardType="number-pad"
                                                                secureTextEntry={pinVisibility}
                                                                onChangeText={(val) => {
                                                                    props.setFieldValue('password', val)
                                                                    props.setFieldTouched('password', true, false);
                                                                    setErrMsg("")
                                                                }}
                                                                style={styles.label2}

                                                            />
                                                            {pinVisibility ?
                                                                <TouchableOpacity onPress={showPin}>
                                                                    <Text style={styles.showTextPin}>SHOW</Text>
                                                                </TouchableOpacity> :
                                                                <TouchableOpacity onPress={showPin}>
                                                                    <Text style={styles.showTextPin}>HIDE</Text>
                                                                </TouchableOpacity>}
                                                        </View>

                                                    </View>
                                                    {props.touched.password && props.errors.password ? (
                                                        <View style={styles.errView}>
                                                            <Text style={styles.errText}>{props.errors.password}</Text>
                                                        </View>) : null}
                                                </View>                    
                                                
                                                    <Btn title="Proceed" onPress={props.handleSubmit} style={styles.submit} />
                                            </View>
                                        )}

                                    </FormikValidator>

                                    <View style={styles.forgotPswdView}>
                                        <View>
                                            <Icon name="help-circle" color="#757575" size={16} />
                                        </View>
                                        <TouchableOpacity style={styles.forgotPswdTitleView} onPress={redirectToForgotPin}>
                                            <Text style={styles.forgotPswTitle}>Forgot your PIN?</Text>
                                        </TouchableOpacity>
                                    </View>
                                </View>

                            </TouchableWithoutFeedback>

                        </View>
                    </KeyboardAwareScrollView>
                </ScrollView>
            </View>

            <Loader isVisible={loader} />

            <RetrievePin
                visibleRetrieve={showRetrieve}
                returnBack={() => setShowRetrieve(false)}
            />
        </View>
    )
};

export default Login;