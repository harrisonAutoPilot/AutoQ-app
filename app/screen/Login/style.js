import { Dimensions, StyleSheet, Platform } from "react-native";
import {
    heightPercentageToDP as hp,
    widthPercentageToDP as wp,
} from 'react-native-responsive-screen';
const {height, width} = Dimensions.get("screen");

export default loginStyles = StyleSheet.create({
    mainBody: {
        backgroundColor: '#fff',
    },
    agentCover: {
        width: wp('100%'),
        height:400,
        flexDirection: 'row',
        justifyContent: 'flex-end',
    },
    agentImg: {
        width: width - 120,
        height: 400,
        resizeMode: 'contain',
        right: wp('-2%'),
    },
    blueImg: {
        width: width - 120,
        height: 400,
        resizeMode: 'cover',
        right: wp('-2%'),
    },
    cover: {
        width: 500,
        height: 450,

    },
    blueCover: {
        width: wp('100%'),
        position: 'absolute',
        top: 0,
        left: -50,

    },
    innerCover: {
        // backgroundColor:'#fff',
        width: 200,
        height: 200,
        position: 'absolute',
        left: wp('18%'),
        top: hp('5%'),
    },
    logoImg: {
        width: 80,
        height: 80,
    },
    textCover: {
        marginTop: 20,
        paddingLeft: 10,
    },
    smText: {
        fontSize: 22,
        fontFamily: "Urbanist-SemiBold",
        lineHeight: 28,
        color: "#fff",
        letterSpacing: 0.2,
    },
    bgText: {
        fontSize: 28,
        fontFamily: "Urbanist-Regular",
        lineHeight: 36,
        color: "#fff",
        letterSpacing: 0.2,
        fontWeight: "900",
        marginTop: 10,
    },
    bottomCover: {
        width: wp('100%'),
        height: hp('60%'),
        borderTopRightRadius: 35,
        borderTopLeftRadius: 35,
        alignItems: 'center',
        backgroundColor: '#fff',
        top:hp('-13%')
    },
    keyboardUp: {
        top: hp('-25%'),
        height: hp('45%'),
    },
    loginTextCover: {
        marginTop: 30,
    },
    loginText: {
        fontSize: 20,
        fontFamily: "Urbanist-SemiBold",
        lineHeight: 28,
        color: "#424242",
        letterSpacing: 0.2,
    },
    submit: {
        marginTop: 20,
        paddingVertical: 16,
        paddingHorizontal: 24,
        backgroundColor: "#3858CF",
        borderRadius: 100,
        zIndex: 2,
    },
    forgotPswTitle: {
        fontSize: 14,
        fontFamily: "Urbanist-SemiBold",
        lineHeight: 20,
        color: "#757575",
        letterSpacing: 0.1,
        textAlign: "center"
    },
    forgotPswdView: {
        marginTop: Dimensions.get("window").height / 15,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center"
    },
    forgotPswdTitleView: {
        marginLeft: 10
    },
    errView: {
        marginTop: 8,
        zIndex: 9000,
    },
    errCover: {
        marginTop: -30,
        position: 'absolute',
        width: wp('80%'),
        alignSelf: 'center'
    },
    inputHolder: {
        borderWidth: 1,
        borderColor: "#BDBDBD",
        paddingVertical: Platform.OS === "ios" ? 20 : 5,
        paddingHorizontal: 10,
        marginTop: Dimensions.get("window").height / 35,
        borderRadius: 5,
    },
    scrollStyle: {
        height: "100%",
        top: hp('-0% '),
        paddingBottom: 250,

    },
    scrollStyleReg: {
        height: hp('90%'),
        // top: hp('-0% '),
        paddingBottom: 180,

    },
    formCover: {
        // height:hp('40%'),
        flex: 1,
        marginTop: -20,
    },
    registerInputHolder: {
        borderWidth: 1,
        borderColor: "#BDBDBD",
        paddingVertical: Platform.OS === "ios" ? 20 : 5,
        paddingHorizontal: 10,
        marginTop: Dimensions.get("window").height / 20,
        borderRadius: 5,
    },
    inputErrHolder: {
        borderColor: "red"
    },
    inputPinHolder: {
        marginTop: 25,
    },
    labelView: {
        position: "absolute",
        top: -7,
        left: 10,
        backgroundColor: "#fff",
        paddingHorizontal: 5,
        zIndex: 2
    },
    label: {
        fontSize: 12,
        fontFamily: "Urbanist-SemiBold",
        lineHeight: 16,
        color: "#616161",
        letterSpacing: 0.2,
    },
    label2: {
        fontSize: 12,
        fontFamily: "Urbanist-Regular",
        lineHeight: 16,
        color: "#616161",
        letterSpacing: 0.2,
        fontWeight: "400",
        width: "70%"
    },
    errText: {
        color: "red",
        fontSize: 12,
        fontFamily: "Urbanist-Regular",
        lineHeight: 16,
        letterSpacing: 0.2,
        fontWeight: "400"
    },
    pinInputView: {
        flexDirection: "row",
        justifyContent: "space-between",
        width: "100%",
        flexWrap: "wrap"
    },
    inputCover: {
        width: wp('100%'),
        padding: 30,
        paddingTop: 10,
        backgroundColor: '#fff'
    },
    inputPinCover: {
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    changeCover: {
        width: 50,
        position: 'absolute',
        right: 2,
        marginTop: 5,
    },
    pinView: {
        justifyContent: "flex-end",

    },
    showTextPin: {
        color: "#9E9E9E",
        fontSize: 11,
        marginTop: Platform.OS === "android" ? 16 : 5,
        fontFamily: "Urbanist-Regular",
        lineHeight: 16,
        letterSpacing: 0.2,
        fontWeight: "400"
    },
    failedResponseView: {
        marginTop: 10,
        borderWidth: 1,
        borderRadius: 8,
        paddingHorizontal: 5,
        backgroundColor: "red"
    },
    createNewAccountView: {
        marginTop: Dimensions.get("window").height / 4.5,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        position: "absolute",
        top: 100,
        alignSelf: "center"
    },
    loginAccountView: {
        marginTop: Dimensions.get("window").height / 50,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        // position: "absolute",
        // top: 70,
        alignSelf: "center"
    },
    createNewAccountText: {
        color: "#3858CF",
        fontSize: 14,
        fontFamily: "Urbanist-Regular",
        lineHeight: 20,
        letterSpacing: 0.1,
        fontWeight: "600",
        textAlign: "center"
    },
    createAccountlabel: {
        fontSize: 14,
        fontFamily: "Urbanist-Regular",
        lineHeight: 20,
        color: "#212121",
        letterSpacing: 0.25,
    },
    loginImg: {
        height: 800,
        width: "100%"
    },
    loginImgView: {
        position: 'absolute',
        top: hp('75%'),
        left: wp('50%')
        // paddingTop: 10
    },
    afriCover: {
        position: 'absolute',
        right: wp('-1%'),
        top: hp('60%'),
        zIndex: 1,
    },
    afriImg: {
        width: 120,
        height: 170,
    },

    topCover: {
        marginTop: 10,
    },
    registerAccountView: {
        marginTop: 15,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        alignSelf: "center"
    },
    scrollStyle: {
        paddingBottom: 50,

    },
});

