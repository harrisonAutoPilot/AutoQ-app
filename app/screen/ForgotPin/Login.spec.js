import React from "react";
import { act } from 'react-test-renderer';
import { fireEvent, waitFor } from '@testing-library/react-native';
import mockAxios from "axios";

import Login from "@Screen/Login";
import Config from "react-native-config";

beforeAll(() => {
    jest.useFakeTimers()
});

jest.mock("axios");


// Render Component 
it('renders screen correctly', async () => {
    const result = renderReducer(<Login />)
    await waitFor(async () => { expect(result).toMatchSnapshot(); })
});

// Show error messages when validation fails
it("shows error messages when both input fields are empty", async () => {
    const { getByTestId, getByText } = renderReducer(<Login />);

    await waitFor(() => {
        fireEvent.press(getByTestId("LoginID"));
    });

    await act(() => {
        getByText("Phone Number is required")
        getByText("Pin is required")
    });
})

//  Show error messages when password validation fails
it("shows an error message when password input field is empty", async () => {
    const { getByTestId, getByText } = renderReducer(<Login />);

    const phoneInput = getByTestId("phone");

    fireEvent.changeText(phoneInput, "2348134091515")

    await waitFor(() => {
        fireEvent.press(getByTestId("LoginID"));
    })

    await act(async () => {
        getByText("Pin is required")
    });
})

//  Show error messages when phone field is empty
it("shows an error message when phone input field is empty", async () => {
    const { getByTestId, getByText } = renderReducer(<Login />);

    const pinInput = getByTestId("password");

    fireEvent.changeText(pinInput, "1234")

    await waitFor(() => {

        fireEvent.press(getByTestId("LoginID"));
    })

    await act(async () => {
        getByText("Phone Number is required")
    });
})

// Show error messages when phone number is less than 13 digits
it("shows an error message when phone number is less than 13 digits", async () => {
    const { getByTestId, getByText } = renderReducer(<Login />);

    const phoneInput = getByTestId('phone');

    fireEvent.changeText(phoneInput, "234813409");

    await waitFor(() => {

        fireEvent.press(getByTestId("LoginID"));
    })

    await act(async () => {
        getByText("Phone Number should be exactly 13 digits");
    });
})

// Show error messages when phone number is more than 13 digits
it("shows an error message when phone number is more than 13 digits", async () => {
    const { getByTestId, getByText } = renderReducer(<Login />);

    const phoneInput = getByTestId('phone');

    fireEvent.changeText(phoneInput, "23481340912111");

    await waitFor(() => {
        fireEvent.press(getByTestId("LoginID"));
    })

    await act(async () => {
        getByText("Phone Number should be exactly 13 digits");
    });
})

// // Login field validation should pass
it("validation passes", async () => {

    const data = {phone: "2348134091515", password: "1234"}

    expect(mockAxios.post).not.toHaveBeenCalled();

    mockAxios.post.mockResolvedValueOnce({data});

   renderReducer(<Login />)

    // const phone = getByTestId('phone');

    // const password = getByTestId('password');

    // fireEvent.changeText(phone, "2348134091515");

    // fireEvent.changeText(password, "1234");

    expect(mockAxios.post).toHaveBeenCalledTimes(1);

//    await waitFor(() => {
//     expect(mockAxios.post).toHaveBeenCalledTimes(1);
//     })

 
    

    // const result =  renderReducer(<Login />)

    // console.log(call)



    // expect(result).toEqual(data);

    // await act(async () => {
        // expect(mockAxios.post).toHaveBeenCalledWith(`${Config.API_URL}/api/v1/login`);
        // expect(mockAxios.post).toHaveBeenCalled()
    // });

    // mockAxios.post.mockImplementationOnce(() => Promise.resolve({data: {phone: "2348134091515", password: "1234"}}));

    // mockAxios.get.mockResolvedValue(Promise.resolve({phone: "2348134091515", password: "1234"}));

   

   

  

   
    // mockAxios.post.mockImplementation(() => Promise.resolve({data: }))

    // await act(async () => {
    //      expect(mockAxios.post).toHaveBeenCalled()
    // })

    // expect(mockAxios.post).toHaveBeenCalledWith("api/v1/login", expect.objectContaining({
    //         phone: "2348134091515",
    //         password: "1234"
    //     }))

})