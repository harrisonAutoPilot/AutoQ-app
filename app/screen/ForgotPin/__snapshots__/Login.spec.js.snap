// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders screen correctly 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
        }
      }
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
      >
        <View
          style={
            Object {
              "borderBottomWidth": 0,
              "flexDirection": "row",
              "marginTop": 20,
              "paddingLeft": 19,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../app/asset/image/left.png",
                }
              }
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "justifyContent": "center",
                "width": 592.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#212121",
                  "fontFamily": "Urbanist-SemiBold",
                  "fontSize": 18,
                  "letterSpacing": 0.1,
                  "lineHeight": 24,
                  "textAlign": "center",
                }
              }
            />
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
    <View
      style={
        Object {
          "marginTop": 20,
          "paddingHorizontal": 20,
          "paddingTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 22.233333333333334,
            "marginTop": -20,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/asset/image/rh_logo.png",
            }
          }
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#212121",
              "fontFamily": "Urbanist-SemiBold",
              "fontSize": 22,
              "letterSpacing": 0.2,
              "lineHeight": 28,
              "textAlign": "center",
              "top": -26.5,
            }
          }
        >
          Login to your Account
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "marginLeft": 37.5,
              "marginRight": 112.5,
              "marginTop": -30,
              "position": "absolute",
              "width": 600,
            }
          }
        />
        <View>
          <View>
            <View>
              <View
                style={
                  Array [
                    Object {
                      "borderColor": "#BDBDBD",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "marginTop": 78.47058823529412,
                      "paddingHorizontal": 10,
                      "paddingVertical": 20,
                    },
                    null,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "left": 10,
                      "paddingHorizontal": 5,
                      "position": "absolute",
                      "top": -10,
                      "zIndex": 2,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#616161",
                        "fontFamily": "Urbanist-SemiBold",
                        "fontSize": 12,
                        "letterSpacing": 0.2,
                        "lineHeight": 16,
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    PHONE NUMBER
                  </Text>
                </View>
                <TextInput
                  autoFocus={true}
                  keyboardType="number-pad"
                  maxLength={13}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="234809XXXXXXX"
                  placeholderTextColor="#757575"
                  style={
                    Object {
                      "color": "#212121",
                      "fontFamily": "Urbanist-Regular",
                      "fontSize": 12,
                      "letterSpacing": 0.2,
                      "lineHeight": 16,
                      "width": "70%",
                    }
                  }
                  testID="phone"
                  value=""
                />
              </View>
            </View>
            <View>
              <View
                style={
                  Array [
                    Object {
                      "borderColor": "#BDBDBD",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "marginTop": 78.47058823529412,
                      "paddingHorizontal": 10,
                      "paddingVertical": 20,
                    },
                    Object {
                      "marginTop": 25,
                    },
                    null,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "left": 10,
                      "paddingHorizontal": 5,
                      "position": "absolute",
                      "top": -10,
                      "zIndex": 2,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#616161",
                        "fontFamily": "Urbanist-SemiBold",
                        "fontSize": 12,
                        "letterSpacing": 0.2,
                        "lineHeight": 16,
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    4 DIGIT PIN
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "flexWrap": "wrap",
                      "justifyContent": "space-between",
                      "width": "100%",
                    }
                  }
                >
                  <TextInput
                    keyboardType="number-pad"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="****"
                    placeholderTextColor="#757575"
                    secureTextEntry={true}
                    style={
                      Object {
                        "color": "#212121",
                        "fontFamily": "Urbanist-Regular",
                        "fontSize": 12,
                        "letterSpacing": 0.2,
                        "lineHeight": 16,
                        "width": "70%",
                      }
                    }
                    testID="password"
                    value=""
                  />
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#9E9E9E",
                          "fontFamily": "Urbanist-Regular",
                          "fontSize": 11,
                          "fontWeight": "400",
                          "letterSpacing": 0.2,
                          "lineHeight": 16,
                          "marginTop": 5,
                        }
                      }
                    >
                      SHOW
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            />
            <View
              style={
                Object {
                  "zIndex": 9,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#3858CF",
                    "borderRadius": 100,
                    "marginTop": 20,
                    "opacity": 1,
                    "paddingHorizontal": 24,
                    "paddingVertical": 16,
                    "zIndex": 2,
                  }
                }
                testID="LoginID"
              >
                <Text
                  style={
                    Object {
                      "color": "#fff",
                      "fontFamily": "Urbanist-SemiBold",
                      "fontSize": 16,
                      "letterSpacing": 0.1,
                      "lineHeight": 24,
                      "textAlign": "center",
                    }
                  }
                >
                  Log In
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 66.7,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": -7.5,
          "top": 907,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/asset/image/afri.png",
          }
        }
        style={
          Object {
            "height": 170,
            "width": 120,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "left": 375,
          "position": "absolute",
          "top": 1000.5,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": 800,
            "width": "100%",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/asset/image/login.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 800,
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "absolute",
              "top": 100,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#212121",
                  "fontFamily": "Urbanist-Regular",
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                }
              }
            >
              Don't have an account yet?
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#3858CF",
                  "fontFamily": "Urbanist-SemiBold",
                  "fontSize": 14,
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "textAlign": "center",
                }
              }
            >
               CREATE NOW
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Modal
      animationType="none"
      deviceHeight={null}
      deviceWidth={null}
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      panResponderThreshold={4}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      statusBarTranslucent={false}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        forwardedRef={[Function]}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        collapsable={false}
        deviceHeight={null}
        deviceWidth={null}
        forwardedRef={[Function]}
        hideModalContentWhileAnimating={false}
        nativeID="animatedComponent"
        onBackdropPress={[Function]}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        panResponderThreshold={4}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        statusBarTranslucent={false}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Object {
              "height": 1334,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              collapsable={false}
              nativeID="animatedComponent"
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../app/asset/image/rh_logo_splashscreen.png",
                }
              }
              style={
                Object {
                  "height": 100,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 100,
                }
              }
            />
          </View>
        </View>
      </View>
    </Modal>
  </View>,
  " ",
]
`;
